pandas/core/base.py

```diff
@@ -15,6 +16,24 @@
                             duplicated='IndexOpsMixin')
 
 
+def _validate_sorter(obj, sorter):
+    """
+    Checks whether or not the implementation of 'searchsorted'
+    for a given class instance 'obj' should accept a non-None
+    'sorter' argument, if one is provided. Raises a
+    NotImplementedError if that is not the case.
+    """
```

Comment:
I don't think I quite follow what you are saying.  What do you mean by "you need to put the actual impl here as well"?  I'm sure what you are saying makes sense in conception, but it is not coming across clearly to me at the moment on how to proceed with regards to the organization.